_posts/en/newsletters/2020-02-05-newsletter.md

```diff
@@ -0,0 +1,279 @@
+
+- **Help test Bitcoin Core 0.19.1rc1:** this upcoming maintenance
+  [release][bitcoin core 0.19.1] includes several bug fixes.
+  Experienced users are encouraged to help test for any regressions or
+  other unexpected behavior.
+
+## News
+
+- **Taproot and tapscript experimentation tool:** Karl-Johan Alm
+  [posted][alm btcdeb] to the Bitcoin-Dev list about an experimental branch of his
+  [btcdeb][] tool that supports creating and executing [taproot][topic
+  taproot] and [tapscript][topic tapscript] outputs using a command line
+  tool named `tap`.  For more information, see his detailed
+  [tutorial][tap tutorial].
+
+{% comment %}<!-- Timeline:
+1/26 15:29 gmaxwell
+https://github.com/bitcoin-core/secp258k1/pull/558#discussion_r371027220
+"shouldn't this present an interface that takes the pubkey as an
+argument,"
+
+1/26 16:05 jonasnick
+"Note that in that case we'll need to add the pubkey as extradata to the
+deterministic nonce function, otherwise calling sign with the wrong
+public key may leak the secret key through an (invalid) signature."
+
+1/27 18:36 #secp256k1
+[6:36:02 pm] <gmaxwell> but it's not really a problem in ed25519 because the only thing it does with the public key is stuffs it into the e hash, and presumably it stuffs the same data into the nonce hash?  (if it doesn't they're pants on head stupid and should be shot)
+[...]
+[6:37:49 pm] <gmaxwell> okay the ed25519 people should be shot, it's an input to e and not an input to their nonce function
+[6:38:02 pm] <gmaxwell> (I just checked)
+
+1/28 01:49 https://moderncrypto.org/mail-archive/curves/2020/001012.html
+-->{% endcomment %}
+
+- **Safety concerns related to precomputed public keys used with schnorr signatures:** [libsecp256k1 PR
+  #558][libsecp256k1 #558] proposes adding [BIP340][]-compatible
+  [schnorr signature][topic schnorr signatures] creation and
+  verification to the libsecp256k1 library used by Bitcoin Core and
+  several other Bitcoin programs.  Since BIP340 requires signatures
+  commit to the public key that will be used to validate the signature,
+  the current proposed signature function uses the private key to derive
+  the necessary public key.  Gregory Maxwell [noted][gmaxwell pubkey]
+  that programs generating signatures will usually already know the
+  appropriate public key, so the function can save CPU time by accepting
+  the public key as a parameter.
+
+    Jonas Nick [replied][nick nonce] that the approach was reasonable,
+    but doing it safely would require including the public key in the
+    data used to create a deterministic nonce.  Otherwise, if the
+    interface is exposed in a way that would allow an attacker to get
+    you to return two signatures created by the same private key but for
+    two different public keys, with all the other data was the same,
+    that would result in nonce reuse which would allow the attacker to
+    derive your private key and steal your bitcoins.  Discussion about
```

Comment:
This is a difficult sentence to follow.

Here is a shot at a simpler sentence:
If an attacker could obtain two signatures created by the same private key for two different public keys, and all of the other data remained the same, you would unknowingly be reusing a nonce and the attacker could derive your private key and steal your funds.

Reply:
Largely took your wording here.